app/controllers/sessions_controller.rb

```diff
@@ -6,29 +6,28 @@ def create
   end
 
   def destroy
+    delete_legacy_cookie
+
     # Let's say goodbye to all the cookies when someone signs out.
     cookies.clear(domain: cookie_domain)
 
-    delete_legacy_cookie
-
     super
   end
 
   private
 
-  def cookie_domain
+  def legacy_cookie_domain
     Rails.env.production? ? ApplicationConfig["APP_DOMAIN"] : nil
   end
 
-  # NOTE: this code is a hotfix, and shall be removed soon (around 2 weeks from deployment)
-  def delete_legacy_cookie
-    domain = cookie_domain
-
-    return unless domain&.starts_with?(".")
+  def cookie_domain
+    ".#{legacy_cookie_domain}"
```

Comment:
I think that eventually, we can just change this to be `".#{ApplicationConfig["APP_DOMAIN"]}"`.